rundoctests.py

```diff
@@ -24,47 +62,50 @@
 for typ in [AlarmInterrupt, SignalError]:
     typ.__module__ = "__main__"
 
+if os.name != 'nt':
+    # Limit stack size to avoid errors in stack overflow doctest
+    stacksize = 1 << 20
+    resource.setrlimit(resource.RLIMIT_STACK, (stacksize, stacksize))
+
+    # Disable core dumps
+    resource.setrlimit(resource.RLIMIT_CORE, (0, 0))
+
+def testfile(file):
+    # Child process
+    try:
+        if sys.platform == 'darwin':
+            from cysignals.signals import _setup_alt_stack
+            _setup_alt_stack()
+        failures, _ = doctest.testfile(file, module_relative=False, optionflags=flags, parser=parser)
+        if not failures:
+            os._exit(0)
+    finally:
+        os._exit(23)
+
+if __name__ == "__main__": # Mandatory for windows cases.
+    success = True
+    for f in filenames:
+        print(f)
+        sys.stdout.flush()
+
+        # Test every file in a separate process (like in SageMath) to avoid
+        # side effects from doctests.
+        p = Process(target=testfile, args=(f,))
+        p.start()
+        p.join(200)
```

Comment:
Maybe to be safer, increase the timeout to 10 minutes or so. And better be explicit and use `join(timeout=600)`. You also need to handle the case of timeout.